Source/NSBrowser.m

```diff
@@ -3410,23 +3530,63 @@ - (void) _performLoadOfColumn: (NSInteger)column
   [sc setDocumentView: matrix];
 
   // Loading is different based upon item/passive/active delegate
-  if (_itemBasedDelegate)
+  if (_itemBasedDelegate == YES) //  && item != nil && tc != nil)
     {
-      // Iterate over the children for the item....
-      for (i = 0; i < rows; i++)
+      NSString *childrenKeyPath = [tc childrenKeyPathForNode: item];
+
+      if (childrenKeyPath != nil)
 	{
-	  id aCell = [matrix cellAtRow: i column: 0];
-	  if (![aCell isLoaded])
+	  NSString *leafKeyPath = [tc leafKeyPathForNode: item];
+	  NSString *keyPath = [self _keyPathForValueBinding];
+
+	  // Iterate over the children for the item....
+	  for (i = 0; i < rows; i++)
 	    {
-	      BOOL leaf = YES;
-	      id val = nil;
-
-	      child = [_browserDelegate browser: self child: i ofItem: item];
-	      leaf = [_browserDelegate browser: self isLeafItem: child];
-	      val = [_browserDelegate browser: self objectValueForItem: child];
-	      [aCell setLeaf: leaf];
-	      [aCell setObjectValue: val];
-	      [aCell setLoaded: YES];
+	      id aCell = [matrix cellAtRow: i column: 0];
+	      if (![aCell isLoaded])
+		{
+		  BOOL leaf = YES;
+		  id val = nil;
+		  NSNumber *leafBool = [child valueForKeyPath: leafKeyPath];
+
+		  child = [children objectAtIndex: i];
+		  leaf = [leafBool boolValue];
+
+		  // If a content values binding is present, it uses that key path,
+		  // but if one isn't it uses the description... per documentation.
+		  if (keyPath != nil)
+		    {
+		      val = [child valueForKeyPath: keyPath];
```

Comment:
@fredkiefer Per our discussion yesterday I am going to try a few things to get the value directly with bindings.  It seems to me that arrangedObjects.value would work if the data structure was a simple array, but given that it is a set of tree nodes with embedded children it doesn't seem quite as straightforward.  It is still, perhaps, that I am not seeing something that is obvious.

Reply:
@fredkiefer Honestly, I have been thinking about this a lot. I don't know if there is a way to simplify it. With the path that is given arrangedObjects.value, this syntax is okay for arrays and simple data structures, but I am not sure it's going to work as we are using NSTreeNode or similar data structures. I agree that bindings should make things easier but I am not sure how to make it easier for US. *sigh*. As implemented this works as expected, but it uses the same "trick" to get the value as NSOutlineView.   Alternatively, I would need to concatenate the value keyPath with arrangedObjects to get the above or, alternatively, get if it's a leaf or other attributes by doing something similar (manipulating the keyPath).

So it seems no matter what I do the solution is slightly messy.   I apologize, a lot of this is me ranting after setting her cogitating this for hours on end.